dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java

```diff
@@ -35,58 +37,82 @@
     @Override
     @CloseDB
     protected final void service(HttpServletRequest req, HttpServletResponse response) throws ServletException, IOException {
-        VelocityRequestWrapper request = new VelocityRequestWrapper(req);
+        final VelocityRequestWrapper request = new VelocityRequestWrapper(req);
         final String uri = CMSUrlUtil.getCurrentURI(request);
-
+        final boolean comeFromSomeWhere = request.getHeader("referer") != null;
+        final User frontEndUser = ((UserWebAPIImpl) WebAPILocator.getUserWebAPI()).getLoggedInFrontendUser(request);
+        request.setRequestUri(uri);
+        final PageMode mode = PageMode.getWithNavigateMode(request);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028984)